src/controller/CurrentFabricRemover.h

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/OperationalSessionSetup.h>
+#include <controller/CHIPDeviceController.h>
+#include <lib/core/CHIPCallback.h>
+
+namespace chip {
+namespace Controller {
+
+typedef void (*OnCurrentFabricRemove)(void * context, NodeId remoteDeviceId, CHIP_ERROR status);
+
+/**
+ * A helper class to remove fabric given some parameters.
+ */
+class CurrentFabricRemover
+{
+public:
+    CurrentFabricRemover(DeviceCommissioner * commissioner) :
+        mCommissioner(commissioner), mOnDeviceConnectedCallback(&OnDeviceConnectedFn, this),
+        mOnDeviceConnectionFailureCallback(&OnDeviceConnectionFailureFn, this)
+    {}
+
+    enum class Step : uint8_t
+    {
+        // Ready to start removing a fabric.
+        kAcceptRemoveFabricStart = 0,
+        // Need to get Current Fabric Index.
+        kReadCurrentFabricIndex,
+        // Need to send Remove Fabric Command.
+        kSendRemoveFabric,
+    };
+
+    /*
+     * @brief
+     *   Try to look up the device attached to our controller with the given
+     *   remote device id and ask it to remove Fabric.
+     *
+     * @param[in] remoteDeviceId The remote device Id
+     * @param[in] callback The callback to call once the remote fabric is completed or not.
```

Comment:
Since `AutoCurrentFabricRemover` relies upon this not to leak instances of itself, it seems like it would be useful to document that:

* callback will never be executed if `CurrentFabricRemove` returns an error
* callback will always be executed if `CurrentFabricRemove` returns CHIP_NO_ERROR

Reply:
I will add additional comment.
"If function returns an error, callback will never be be executed. Otherwise, callback will always be executed."